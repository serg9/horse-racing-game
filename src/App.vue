<template>
  <div class="app">
    <HorseRacingGame />
  </div>
</template>

<script setup>
import HorseRacingGame from './views/HorseRacingGame.vue';
</script>