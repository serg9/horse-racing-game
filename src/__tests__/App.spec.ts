import { describe, it, expect } from 'vitest'
import { shallowMount, mount } from '@vue/test-utils'
import App from '../App.vue'

describe('App.vue', () => {
  it('mounts without crashing (smoke test)', () => {
    const wrapper = shallowMount(App)
    expect(wrapper.exists()).toBe(true)
  })

  it('renders the root container with class "app"', () => {
    const wrapper = shallowMount(App)
    const root = wrapper.find('div.app')
    expect(root.exists()).toBe(true)
  })

  it('includes HorseRacingGame child (stubbed in shallowMount)', () => {
    const wrapper = shallowMount(App)
    // Vue Test Utils shallowMount auto-stubs child component imported in App
    const childStub = wrapper.find('horse-racing-game-stub')
    expect(childStub.exists()).toBe(true)
  })

  it('can mount with an explicit stub for HorseRacingGame', () => {
    const wrapper = mount(App, {
      global: {
        stubs: {
          HorseRacingGame: {
            name: 'HorseRacingGame',
            template: '<section data-test="horse-game-stub" />'
          }
        }
      }
    })

    expect(wrapper.find('[data-test="horse-game-stub"]').exists()).toBe(true)
  })

  it('matches the basic HTML structure snapshot', () => {
    const wrapper = shallowMount(App)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

