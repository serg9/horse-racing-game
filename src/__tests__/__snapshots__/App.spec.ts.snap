// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > matches the basic HTML structure snapshot 1`] = `
"<div class="app">
  <horse-racing-game-stub></horse-racing-game-stub>
</div>"
`;
